OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs

```diff
@@ -78,7 +80,14 @@ public void StartGame(Arguments args)
 			var replay = args != null ? args.GetValue("Launch.Replay", null) : null;
 			if (!string.IsNullOrEmpty(replay))
 			{
-				Game.JoinReplay(replay);
+				var replayMeta = ReplayMetadata.Read(replay);
```

Comment:
You will also need to reorder the checks inside `CheckReplayCompatibility` to first check for `replayMeta == null`, then `version == null`, then `mod == null`.